operator/internal/lifecycle/helpers.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package lifecycle
+
+import (
+	"fmt"
+	"slices"
+	"sort"
+	"strconv"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func isStatefulSet(object client.Object) bool {
+	gvk := object.GetObjectKind().GroupVersionKind()
+	return gvk.Group == appsv1.SchemeGroupVersion.Group &&
+		gvk.Version == appsv1.SchemeGroupVersion.Version &&
+		gvk.Kind == "StatefulSet"
+}
+
+// getGroupVersionKind gets a GVK for an object based on all
+// GVKs registered with a runtime scheme.
+func getGroupVersionKind(scheme *runtime.Scheme, object client.Object) (*schema.GroupVersionKind, error) {
+	kinds, _, err := scheme.ObjectKinds(object)
+	if err != nil {
+		return nil, fmt.Errorf("fetching object kind: %w", err)
+	}
+	if len(kinds) == 0 {
+		return nil, fmt.Errorf("unable to determine object kind")
+	}
+
+	gvk := kinds[0]
+	return &schema.GroupVersionKind{
+		Group:   gvk.Group,
+		Version: gvk.Version,
+		Kind:    gvk.Kind,
+	}, nil
+}
+
+// sortCreation sorts a list of objects by their creation timestamp
```

Comment:
nit: Technically this is "sort objects by (creation timestamp, name)" it's misleading to not mention that.